stm32_base/hal/usart_stm32/usart_device_stm32.c

```diff
@@ -729,6 +739,15 @@ void USART2_IRQHandler(void)
 //		xQueueSendFromISR( xUsart2Rx, &cChar, &xTaskWokenByPost );
     }
 
+    if (USART_GetITStatus(USART2, USART_IT_ORE_RX) != RESET) {
```

Comment:
Dude... let the optimizer handle that.  That is what it is there for.  Writing clean small scope code should be your concern.  If you are thinking about that here, you are over optimizing.  3 cycles to push frames and do a call is negligible considering the speed of this processor.